src/python/WMCore/MicroService/MSPileup/MSPileupTasks.py

```diff
@@ -100,17 +105,131 @@ def pileupSizeTask(self):
+        - add new ruleIds to pileup document
+
+        :return: None
+        """
+        spec = {'active': True}
+        docs = self.mgr.getPileup(spec)
+        for doc in docs:
+            self.logger.info("partialPileupTask process %s", doc)
+            fraction = doc['containerFraction']
+
+            # check previous fraction value and proceed only if has changed
+            previousFraction = 1.0
+            transition = doc.get('transition', [])
+            if transition:
+                previousFraction = transition[-1]['containerFraction']
+            if previousFraction == fraction:
+                # do nothing since there is no transition, i.e. skip this record
+                self.logger.info("Pileup name %s has no container fractions changes", doc['pileupName'])
+                continue
+
+            # get rucio DIDs (datasets, in CMS name blocks) for our pileup document as following:
+            # - when container fraction is decreasing: new container is a subset of customName
+            # - when container fraction is increasing: new container is a superset of customName + subset of pileupName
+
+            # usage of block names defined in this logic:
+            # https://github.com/dmwm/WMCore/pull/11807#pullrequestreview-1786778783
+            pname = doc['pileupName']
+            blockNames = self.rucioClient.getBlocksInContainer(pname)
+            if previousFraction < fraction:
+                # if it increasing
+                # use EVERY single block defined in the custom dataset plus blocks from the standard pileup
+                cname = doc.get('customName', '')
+                if cname:
+                    blockNames = self.rucioClient.getBlocksInContainer(cname) + blockNames
+            else:
+                # if it is decreasing
+                # instead of using random blocks from the standard pileup dataset (pileupName)
+                # we should use it from the custom dataset (customName) - if existent
+                cname = doc.get('customName', '')
+                if cname:
+                    blockNames = self.rucioClient.getBlocksInContainer(cname)
+
+            # get portion of DIDs based on ceil(containerFraction * num_rucio_datasets)
+            portion = blockNames[:math.ceil(fraction * len(blockNames))]
+            self.logger.info("For pileup document %s take portion %s", doc, portion)
+
+            # create new container DID in Rucio for our custom name
+            self.rucioClient.createContainer(doc['customName'], scope=self.customRucioScope)
+
+            # we call attachDIDs Rucio wrapper API with our set of DIDs
+            newRules = []
+            for rse in doc['currentRSEs']:
+                self.rucioClient.attachDIDs(rse, doc['customName'], portion, scope=self.customRucioScope)
+
+                # create new rule for custom DID using pileup document rse
+                newRules += self.rucioClient.createReplicationRules(portion, rse)
```

Comment:
Replication rule needs to be created for the custom pileup, not for all the block names. So please replace `portion` by `doc['customName']`.

BTW, where is `createReplicationRules` defined? I think you mixed up the method name here.

Reply:
ok, done. The `createReplcationRule` are defined here  https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Services/Rucio/Rucio.py#L444 I fixed the function name though, i.e. `createReplicationRules` to `createReplicationRule`.

Reply:
See my comment above, you are still creating rules for the wrong DID object.